fastapi/openapi/docs.py

```diff
@@ -96,8 +97,30 @@ def get_redoc_html(
     </style>
     </head>
     <body>
-    <redoc spec-url="{openapi_url}"></redoc>
+    <redoc id="redoc-container"></redoc>
     <script src="{redoc_js_url}"> </script>
+    <script src="{mermaid_js_url}"> </script>
+    <script type="application/javascript">
+        mermaid.initialize({{startOnLoad: false}});
+        let id = 0;
+        Redoc.init("{openapi_url}", {{
+            theme: {{
+                extensionsHook: (name, props) => {{
+                    if (name !== "Markdown" || props.html === undefined) return "";
+                    var markdown = document.createElement("div");
+                    markdown.innerHTML = props.html;
+                    var mermaidNodes = markdown.querySelectorAll("pre > code.language-mermaid");
+                    for (const mermaidNode of mermaidNodes) {{
+                        mermaid.render(`mermaid-${{id++}}`, mermaidNode.textContent, svg => {{
+                            mermaidNode.parentNode.outerHTML = svg;
+                        }});
+                    }}
+                    props.dangerouslySetInnerHTML.__html = markdown.innerHTML;
+                    return "";
+                }}
+            }}
+        }}, document.getElementById("redoc-container"));
+    </script>
```

Comment:
Snippet is taken from https://github.com/Redocly/redoc/issues/266#issuecomment-631252318